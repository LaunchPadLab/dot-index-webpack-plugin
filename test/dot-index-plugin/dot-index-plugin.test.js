const webpack = require('webpack')
const path = require('path')
const DotIndexPlugin = require('../../src')
const fs = require('fs-extra')

const to = relPath => path.resolve(__dirname, relPath)

test('Generates dot index files', end => {
  
  const config = {
    entry: to('./test-input'),
    output: {
      path: to('./test-output'),
      filename: '[name].js'
    },
    resolve: {
      mainFiles: ['index', '.index'],
    },
    plugins: [
      new DotIndexPlugin({ path: to('./test-input') })
    ]
  }

  webpack(config, (err, stats) => {
    expect(err).toEqual(null)
    const modules = stats.toJson().modules
    expect(modules.length).toEqual(4)
    const indexFileContent = modules[0].source
    expect(indexFileContent).toMatchSnapshot()
    end()
  })

})

afterAll(() => {
  fs.removeSync(to('./test-input/.index.js'))
  fs.removeSync(to('./test-output'))
})