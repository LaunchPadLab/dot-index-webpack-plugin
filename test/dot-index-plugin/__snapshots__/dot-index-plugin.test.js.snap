// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accepts formatExports argument 1`] = `
"// This index file was auto-generated by dot-index-webpack-plugin.
// To overwrite it, simply add an index.js file of your own.

export { default as MYCOMPONENT } from './MyComponent';
export { default as MYKEBABCASEDFUNCTION } from './my-kebab-cased-function';
export { default as MYFUNCTION } from './myFunction';"
`;

exports[`Generates dot index files 1`] = `
"// This index file was auto-generated by dot-index-webpack-plugin.
// To overwrite it, simply add an index.js file of your own.

export { default as MyComponent } from './MyComponent';
export { default as myKebabCasedFunction } from './my-kebab-cased-function';
export { default as myFunction } from './myFunction';"
`;
